app/renderer/components/Inspector/GestureEditor.js

```diff
@@ -153,137 +159,142 @@ class GestureEditor extends Component {
     }
     return newPointers;
   }
 
-  pixelsToPercentage (pixel, isX) {
+  getDefaultMoveDuration (ticks, tickId, x2, y2) {
     const {width, height} = this.props.windowSize;
-    if (!isNaN(pixel)) {
-      if (isX) {
-        return parseFloat(((pixel / width) * 100).toFixed(1), 10);
-      } else {
-        return parseFloat(((pixel / height) * 100).toFixed(1), 10);
+    const {coordType} = this.state;
+    const {PERCENTAGES} = COORD_TYPE;
+    const ticksExceptCurrent = ticks.filter((tick) => tick.id !== tickId);
+    const prevPointerMoves = [];
+    for (const tick of ticksExceptCurrent) {
+      if (tick.type === POINTER_MOVE && tick.x !== undefined && tick.y !== undefined) {
+        prevPointerMoves.push({x: tick.x, y: tick.y});
       }
-    } else {
-      return undefined;
     }
-  }
-
-  percentageToPixels (percentage, isX) {
-    const {width, height} = this.props.windowSize;
-    if (!isNaN(percentage)) {
-      if (isX) {
-        return Math.round(width * (percentage / 100));
-      } else {
-        return Math.round(height * (percentage / 100));
+    const len = prevPointerMoves.length;
+    if (len !== 0) {
+      const obj = {x1: prevPointerMoves[len - 1].x, y1: prevPointerMoves[len - 1].y, x2, y2};
+      if (coordType === PERCENTAGES) {
+        obj.x1 = percentageToPixels(obj.x1, true, width, height);
+        obj.y1 = percentageToPixels(obj.y1, false, width, height);
+        obj.x2 = percentageToPixels(obj.x2, true, width, height);
+        obj.y2 = percentageToPixels(obj.y2, false, width, height);
       }
+      const eq = (val1, val2) => Math.sqrt((val1 ^ 2) + (val2 ^ 2));
+      const xDiff = obj.x2 - obj.x1;
+      const yDiff = obj.y2 - obj.y1;
+      const maxScreenSize = eq(width, height);
+      const lineLength = eq(xDiff, yDiff);
+      const lineLengthPct = lineLength / maxScreenSize;
+      return Math.round(lineLengthPct * DEFAULT_DURATION_TIME);
     } else {
-      return undefined;
+      return 0;
     }
   }
 
   // Update tapped coordinates within local state
   updateCoordinates (tickKey, updateX, updateY) {
     if (updateX && updateY) {
+      const {width, height} = this.props.windowSize;
       const {pointers, coordType} = this.state;
       const currentPointer = pointers.find((pointer) => pointer.id === tickKey[0]);
-      const currentTick = (currentPointer.ticks).find((tick) => tick.id === tickKey);
+      const currentTick = currentPointer.ticks.find((tick) => tick.id === tickKey);
       const x = parseFloat(updateX, 10);
       const y = parseFloat(updateY, 10);
       if (coordType === COORD_TYPE.PERCENTAGES) {
-        currentTick.x = this.pixelsToPercentage(x, true);
-        currentTick.y = this.pixelsToPercentage(y, false);
+        currentTick.x = pixelsToPercentage(x, true, width, height);
+        currentTick.y = pixelsToPercentage(y, false, width, height);
       } else {
         currentTick.x = x;
         currentTick.y = y;
       }
+
+      if (!currentTick.duration) {
```

Comment:
this will probably update the duration if the user has entered "0" manually, right? ideally we would not update the duration if the user has interacted with the field at all.